documentation/docs/requirements.md

```diff
@@ -0,0 +1,85 @@
+# Overview
+The _Modu_ tool makes use of _Poetry_ to install the majority of the requirements needed. To run the _Modu_ tool itself, these are the only dependencies needed. 
+
+As part of the project was implementing an existing quality model, we have a couple of other dependencies that will need to be installed to run our re-creation of the quality model that Code Climate makes use of
+
+## Installing poetry and dependencies
+### Python
+Our tool will require a user to have a python version of 3.10 or higher. Please ensure that you are making use of an appropriate python version
+### Python Poetry
+As mentioned before, _Modu_ makes use of _Poetry_ to manage the majority of our dependencies. Please refer to the poetry documentation on how to install it: https://python-poetry.org/docs/
+### Tree-sitter
+_Tree-sitter_ is used for generating abstract syntax tree that can be used for querying specific node types. _Tree-sitter_ requires specifically built grammar types for each coding language analyzed. These grammars can be generated by doing the following: 
+
+Ensure that you are in the **tool** folder. 
+```
+tool/
+    ...
+    documentation/
+    figures/
+    modu/
+    ...
+```
+
+Run the following commands:
+Generation of build and grammars folder:
+```sh
+mkdir build grammars
+```
+Clone the different grammars into grammars/:
+```sh
+cd grammars && git clone https://github.com/tree-sitter/tree-sitter-python && git clone https://github.com/tree-sitter/tree-sitter-c-sharp && git clone https://github.com/tree-sitter/tree-sitter-java && cd -
+```
+Build the grammar file:
+```sh
+poetry run python modu/build_treesitter.py
+```
+
+**Adding new languages**
```

Comment:
You will need to do more. `SourceRepository` will also need to be extended (with a tree sitter language object, a tree sitter parser object, this might not be an exhaustive list).

Reply:
done